### ISSUE TITLE
500 error on project deletion

### ISSUE THREAD

**Issue by TheSerapher** [gitlabhq/gitlabhq#9731]
Testing the new Gitlab version and I have encountered an issue when creating a project, then when seeing the initial project page with instruction hitting the REMOVE button. The project is moved properly by sidekick but the redirect in https://github.com/gitlabhq/gitlabhq/blob/v8.0.4/app/controllers/projects_controller.rb#L113-L117 does not happen properly and returns a `500`:

```
Started DELETE "/root/test" for 127.0.0.1 at 2015-10-14 07:43:33 +0000
Processing by ProjectsController#destroy as HTML
  Parameters: {"authenticity_token"=>"[FILTERED]", "namespace_id"=>"root", "id"=>"test"}
Completed 500 Internal Server Error in 1227ms (ActiveRecord: 45.6ms)

NoMethodError (undefined method `include?' for nil:NilClass):
  app/controllers/projects_controller.rb:113:in `destroy'
```

Not sure what would be causing referred to be empty here, but if you need me to test anything let me know.

**EDIT** This also happens when trying to delete through the projects settings page. Gitlab 8.0.4 is my version installed from Omnibus.

**Comment by stanhu** [comment]
It sounds like we need to add a check for an empty referrer. I'm wondering how this happens in the first place. Are you accessing your site via HTTPS or HTTP? Do you have a browser/plugin that removes the referrer header?

Here's a quick way to test. Open the Inspect Element page in your browser and delete the project.  You should see something like this:

![image](https://cloud.githubusercontent.com/assets/963826/10546739/a19902d8-73e4-11e5-9ea4-6bcfdebc4100.png)

**Comment by TheSerapher** [comment]
It could be a loadbalancer doing that. I will check as soon as I am back to work in about two weeks ;-)

**Comment by ssbarnea** [comment]
I see this closed but now Milestone being set which makes me wonder in which release I should see it fixed.

**Comment by Razer6** [comment]
This was included in 8.1
